### PR TITLE
More detailed documentation for uplc.md

### PR DESCRIPTION
**Pull request by ChristianoBraga**
Changed regexp for version. Added explanations throughout the spec. Builtins are not properly handled yet.

### CODE REVIEW

File: uplc.md [hunk 1]
```diff
@@ -27,7 +44,8 @@ module PLUTUS-CORE-LEXICAL-GRAMMAR
    syntax BuiltinName  ::= Name                                  // builtin term name
 // syntax Integer      ::= r"[+-]?[0-9]+"                [token] // int
    syntax ByteString   ::= r"#([a-fA-F0-9][a-fA-F0-9])+" [token] // hex string
-   syntax Version      ::= r"[0-9]+(.[0-9]+)*"           [token] // version
+// syntax Version      ::= r"[0-9]+(.[0-9]+)*"           [token] // version
```
Comment (SchmErik) [inline uplc.md:47]: We can just remove this line. If you want to keep track of it, maybe put it in the comment starting on line 18

File: uplc.md [hunk 2]
```diff
@@ -56,71 +74,186 @@ module UNTYPED-PLUTUS-CORE-GRAMMAR
                   | "(" "lam" Var Term ")"              // lambda abstraction
                   | "(" "delay" Term ")"                // delay execution of a term
 
+   syntax TermList ::= ".TermList" | Term "" TermList
```
Comment (SchmErik) [inline uplc.md:77]: I see you left a "" here. I think this should be removed..

File: uplc.md [hunk 3]
```diff
@@ -56,71 +74,186 @@ module UNTYPED-PLUTUS-CORE-GRAMMAR
...
+```
+
+## AFrame
+
+Frames are used to record an evaluation context. For instance, an
+application term `[M N]` is carried on by first evaluating `M` and
+then `N` is evaluated. Therefore, before the evaluation of `M` starts,
+the frame `[_ N]`, called right application, is pushed onto the stack
+to record the fact that `N` must be evaluated next. The left
+application frame `[ AClosure _]` has a similar mening. The evaluation
```
Comment (SchmErik) [inline uplc.md:136]: change "mening" to "meaning"

File: uplc.md [hunk 4]
```diff
@@ -132,21 +265,37 @@ module UNTYPED-PLUTUS-CORE-CEK
        
   rule <k> (delay M:Term) => M </k>
        <stack> ... (ListItem(Force) => .List) </stack>
+```
+
+### Builtins
+
+TO: Builtins are not being handled properly yet.
```
Comment (SchmErik) [inline uplc.md:272]: TO: => TODO:

**Review by SchmErik — CHANGES_REQUESTED** [review]
I left a few comments

File: uplc.md (hunk 1 above)
Comment (ChristianoBraga) [reply uplc.md:47]: OK

File: uplc.md (hunk 2 above)
Comment (ChristianoBraga) [reply uplc.md:77]: This is part of the syntax. It´s juxtaposition. But I am experimenting with ways to handle this. Using syntactic lists generates ambiguities.

File: uplc.md (hunk 3 above)
Comment (ChristianoBraga) [reply uplc.md:136]: OK

File: uplc.md (hunk 4 above)
Comment (ChristianoBraga) [reply uplc.md:272]: OK

**Pull request closed without merging**